dali/operators/decoder/audio/generic_decoder.cc

```diff
@@ -197,14 +212,20 @@ struct GenericAudioDecoder<SampleType>::Impl {
     if (!sound_) {
       throw DALIException(make_string("Failed to open encoded data: ", sf_strerror(sound_)));
     }
-
-    ret.length = sf_info_.frames;
-    ret.channels = sf_info_.channels;
-    ret.sample_rate = sf_info_.samplerate;
-    ret.channels_interleaved = true;
-    return ret;
+    return GetAudioMetadata(sf_info_);
   }
 
+  AudioMetadata OpenFromFileImpl(const std::string &filepath) {
+    DALI_ENFORCE(!filepath.empty(), "filepath is empty");
+    sf_info_ = {};
+    sf_info_.format = 0;
```

Comment:
Do you need to set it to 0 while you have `sf_info_ = {}` already?

Reply:
This was copied from OpenImpl.

sf_info_ is just a C-style struct. It doesn't have any constructor:
```
struct SF_INFO
{	sf_count_t	frames ;		/* Used to be called samples.  Changed to avoid confusion. */
	int			samplerate ;
	int			channels ;
	int			format ;
	int			sections ;
	int			seekable ;
} ;

typedef	struct SF_INFO SF_INFO ;
```

Reply:
But I think `sf_info_ = {}` already sets it content to 0, so I'm not sure if you need to do that again for any of its fields.